airflow/contrib/hooks/gcp_cloudml_hook.py

```diff
@@ -0,0 +1,166 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import logging
+import random
+import time
+from airflow import settings
+from airflow.contrib.hooks.gcp_api_base_hook import GoogleCloudBaseHook
+from apiclient.discovery import build
+from apiclient import errors
+from oauth2client.client import GoogleCredentials
+
+logging.getLogger('GoogleCloudML').setLevel(settings.LOGGING_LEVEL)
+
+
+def _retry_with_exponential_delay(request, max_n, is_done_func, is_error_func):
```

Comment:
Thanks Chris. The purpose of retry here is slightly different. Some of our http calls (for example for creating a `Version`, return a long running `[Operation](https://cloud.google.com/ml-engine/reference/rest/Shared.Types/ListOperationsResponse#Operation)` which the user needs to keep polling to determine when it's done. This function is to avoid polling every `n` seconds which might hit user's quota restrictions. In the case of errors we mostly don't want to add extra retry logic on top of existing generic logic (except for the `429` status case which can be retried).